addons/account/models/account_move.py

```diff
@@ -854,7 +854,11 @@ def auto_reconcile_lines(self):
         elif self._context.get('skip_full_reconcile_check') == 'amount_currency_only':
             currency = self._context.get('manual_full_reconcile_currency')
 
-        self.env['account.partial.reconcile'].create({
+        skip_full_reconcile_check = self._context.get('skip_full_reconcile_check')
+        if not sm_debit_move.currency_id and not sm_credit_move.currency_id:
+            skip_full_reconcile_check = True
+
+        self.env['account.partial.reconcile'].with_context(skip_full_reconcile_check=skip_full_reconcile_check).create({
```

Comment:
I feel like this hunk should be where the reconcile() method is called, at first... because that's where you know the use case you're dealing with, and there is no side effect possible.

so, what about patching in account_invoice.py, in register_payment()
```diff
-        return (line_to_reconcile + payment_line).reconcile(writeoff_acc_id, writeoff_journal_id)
+        (line_to_reconcile + payment_line).with_context(skip_full_reconcile_check=True).reconcile(writeoff_acc_id, writeoff_journal_id)
+        return  (line_to_reconcile + payment_line).compute_full_after_batch_reconcile()
```

?